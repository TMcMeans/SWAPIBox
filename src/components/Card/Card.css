.info {
  color: #eabc01;
}

.card {
  border: 2px solid #feda4a;
  border-radius: 10%;
  height: 380px;
  position: relative;
  transition: 0.3s;
}

.list-container {
  margin-top: 160px;
}

ul {
  list-style-type: none;
  padding-left: 0;
  text-align: center;
}

.fave-button,
input[type='submit'],
input[type='reset'] {
  background: none;
  border: none;
  padding: 0;
  cursor: pointer;
  outline: inherit;
}

.button-div {
  display: flex;
  align-items: flex-end;
  flex-direction: column;
  padding: 10px 10px 0 0;
}

.fa-star {
  color: #feda4a;
  font-size: 30px;
  background-color: none;
}
/* 
.fave-button:hover .one,
.fave-button .two {
    display: none;
}
.fave-button:hover .two{
    display: inline;
} */

.info-section {
  border-top: 2px solid yellow;
  text-align: left;
  padding: 5px;
  background: black;
  height: 50%;
  opacity: 1;
}

.card::after {
  content: '';
  background-image: url(../../r2d2.png);
  opacity: 0.6;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  position: absolute;
  z-index: -1;
  background-size: cover;
  border-radius: 10%;
}

.card:hover {
  cursor: pointer;
  transform: scale(1.03);
}
