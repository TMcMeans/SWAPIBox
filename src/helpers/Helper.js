export const getRandomFilm = filmArray => {
  const randomNumber = Math.floor(Math.random() * 6);
  return filmArray[randomNumber];
};

export const cleanPeopleData = async peopleArray => {
  const people = peopleArray.map(async person => {
    const { homeworld, species } = person;

    const fetches = [fetch(homeworld), fetch(species)];
    const response = await Promise.all(fetches);

    const parsedResponses = [response[0].json(), response[1].json()];

    const parsedData = await Promise.all(parsedResponses);

    const homeworldObj = parsedData[0];
    const speciesObj = parsedData[1];

    return {
      name: person.name,
      homeworld: homeworldObj.name,
      homePop: homeworldObj.population,
      species: speciesObj.name,
      isFavorite: false
    };
  });
  return Promise.all(people);
};

export const cleanVehicleData = async vehiclesArray => {
  const vehicles = await vehiclesArray.map(async vehicle => {
    return {
      name: vehicle.name,
      model: vehicle.model,
      class: vehicle.vehicle_class,
      passengers: vehicle.passengers,
      isFavorite: false
    };
  });
  return Promise.all(vehicles);
};

export const cleanPlanetsData = async planetsArray => {
  const planets = await planetsArray.map(async planet => {
    const { residents } = planet;
    const promisedPeople = residents.map(async resident => {
      const response = await fetch(resident);
      return response.json();
    });

    const resolvedResidents = await Promise.all(promisedPeople);

    return {
      name: planet.name,
      terrain: planet.terrain,
      population: planet.population,
      climate: planet.climate,
      residents: resolvedResidents,
      isFavorite: false
    };
  });
  return Promise.all(planets);
};
