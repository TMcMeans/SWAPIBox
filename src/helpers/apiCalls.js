export const fetchFilms = async () => {
  try {
    const url = 'https://swapi.co/api/films/';
    const response = await fetch(url);
    const films = await response.json();
    return films;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchPeople = async () => {
  try {
    const url = 'https://swapi.co/api/people/';
    const response = await fetch(url);
    const people = await response.json();
    return people;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchVehicles = async () => {
  try {
    const url = 'https://swapi.co/api/vehicles/';
    const response = await fetch(url);
    const vehicles = await response.json();
    return vehicles;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchPlanets = async () => {
  try {
    const url = 'https://swapi.co/api/planets/';
    const response = await fetch(url);
    const planets = await response.json();
    return planets;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};
