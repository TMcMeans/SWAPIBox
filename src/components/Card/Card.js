import React, { Component } from 'react';
import PropTypes from 'prop-types'
import './Card.css';

class Card extends Component {
  render() {
    const { cardData, toggleFavorite } = this.props;
    const card = cardData;
    const keys = Object.keys(cardData);
    const listItems = keys.map(listItem => {
      if (listItem === 'residents') {
        if (cardData[listItem].length > 0) {
          const residents = cardData[listItem].join(', ');
          return (
            <li className="temp">
              {listItem}: {residents}
            </li>
          );
        } else if (cardData[listItem].length === 0) {
          return <li className="temp">{listItem}: none</li>;
        }
      }
      if (listItem !== 'isFavorite' && listItem !== 'category') {
        return (
          <li className="temp">
            {listItem}: {cardData[listItem]}
          </li>
        );
      }
    });

    return (
      <div className="card">
        <button onClick={() => toggleFavorite(card)}>
          {!card.isFavorite ? 'favorite' : 'unfavorite'}
        </button>
        <ul className="temp">{listItems}</ul>
      </div>
    );
  }
}

Card.PropTypes = {
  cardData: PropTypes.objectOf,
  key: PropTypes.string,
  id: PropTypes.string,
  toggleFavorite: PropTypes.func
}

export default Card;
