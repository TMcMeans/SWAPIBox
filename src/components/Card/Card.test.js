import React from 'react';
import { shallow } from 'enzyme';

import Card from './Card';
import { mockData } from '../../helpers/mockData';

describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockCardData = {
      name: 'Luke Skywalker',
      homeworld: 'Tatooine',
      population: '200000',
      species: 'Human',
      isFavorite: false,
      category: 'people'
    };

    const mockFunc = jest.fn();
    const wrapper = shallow(
      <Card cardData={mockCardData} toggleFavorite={mockFunc} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
