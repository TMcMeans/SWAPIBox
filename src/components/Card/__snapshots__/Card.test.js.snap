// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
      Object {
        "category": "people",
        "homeworld": "Tatooine",
        "isFavorite": false,
        "name": "Luke Skywalker",
        "population": "200000",
        "species": "Human",
      }
    }
    toggleFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
        >
          favorite
        </button>,
        <ul
          className="temp"
        >
          <li
            className="temp"
          >
            name
            : 
            Luke Skywalker
          </li>
          <li
            className="temp"
          >
            homeworld
            : 
            Tatooine
          </li>
          <li
            className="temp"
          >
            population
            : 
            200000
          </li>
          <li
            className="temp"
          >
            species
            : 
            Human
          </li>
        </ul>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "favorite",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="temp"
            >
              name
              : 
              Luke Skywalker
            </li>,
            <li
              className="temp"
            >
              homeworld
              : 
              Tatooine
            </li>,
            <li
              className="temp"
            >
              population
              : 
              200000
            </li>,
            <li
              className="temp"
            >
              species
              : 
              Human
            </li>,
            undefined,
            undefined,
          ],
          "className": "temp",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "name",
                ": ",
                "Luke Skywalker",
              ],
              "className": "temp",
            },
            "ref": null,
            "rendered": Array [
              "name",
              ": ",
              "Luke Skywalker",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "homeworld",
                ": ",
                "Tatooine",
              ],
              "className": "temp",
            },
            "ref": null,
            "rendered": Array [
              "homeworld",
              ": ",
              "Tatooine",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "population",
                ": ",
                "200000",
              ],
              "className": "temp",
            },
            "ref": null,
            "rendered": Array [
              "population",
              ": ",
              "200000",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "species",
                ": ",
                "Human",
              ],
              "className": "temp",
            },
            "ref": null,
            "rendered": Array [
              "species",
              ": ",
              "Human",
            ],
            "type": "li",
          },
          undefined,
          undefined,
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
          >
            favorite
          </button>,
          <ul
            className="temp"
          >
            <li
              className="temp"
            >
              name
              : 
              Luke Skywalker
            </li>
            <li
              className="temp"
            >
              homeworld
              : 
              Tatooine
            </li>
            <li
              className="temp"
            >
              population
              : 
              200000
            </li>
            <li
              className="temp"
            >
              species
              : 
              Human
            </li>
          </ul>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "favorite",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="temp"
              >
                name
                : 
                Luke Skywalker
              </li>,
              <li
                className="temp"
              >
                homeworld
                : 
                Tatooine
              </li>,
              <li
                className="temp"
              >
                population
                : 
                200000
              </li>,
              <li
                className="temp"
              >
                species
                : 
                Human
              </li>,
              undefined,
              undefined,
            ],
            "className": "temp",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "name",
                  ": ",
                  "Luke Skywalker",
                ],
                "className": "temp",
              },
              "ref": null,
              "rendered": Array [
                "name",
                ": ",
                "Luke Skywalker",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "homeworld",
                  ": ",
                  "Tatooine",
                ],
                "className": "temp",
              },
              "ref": null,
              "rendered": Array [
                "homeworld",
                ": ",
                "Tatooine",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "population",
                  ": ",
                  "200000",
                ],
                "className": "temp",
              },
              "ref": null,
              "rendered": Array [
                "population",
                ": ",
                "200000",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "species",
                  ": ",
                  "Human",
                ],
                "className": "temp",
              },
              "ref": null,
              "rendered": Array [
                "species",
                ": ",
                "Human",
              ],
              "type": "li",
            },
            undefined,
            undefined,
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
